cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminOpHTTPHandler.java

```diff
@@ -193,4 +201,13 @@ public void upgrade(HttpRequest request, HttpResponder responder,
   private String getAdminOpErrorMessage(String opName, String instanceName) {
     return String.format("Error executing admin operation %s for dataset instance %s", opName, instanceName);
   }
+
+  // propagate userid from the HTTP Request in the current thread
+  private void propagateUserId(HttpRequest request) {
```

Comment:
I'm guessing its too late for 3.5.0, but are there plans to add some sort of hook for each method that does this type of stuff?